### PR TITLE
fix(kit): Add missing `TUI_COUNTRIES_MASKS` for Overseas France

### PR DESCRIPTION
**Pull request by VictorienTardif**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

People from Overseas France are not able to pick their phone number code.

## What is the new behavior?

Phone number code for Overseas France has been added to the `TUI_COUNTRIES_MASKS`.
Fixes https://github.com/Tinkoff/taiga-ui/issues/3102

Also (feel free to remove this commit), a leading 0 has been added to all french `TUI_COUNTRIES_MASKS`. 
French numbers always start with a 0 and although it is valid to omit it when giving the telephone area code, some services do not work without it. This is the case with Firebase, for example, which does not send SMS if the 0 is missing.
(`+33 0# ## ## ## ##` would have been better but this mask does not work while `+33(0) # ## ## ## ##` does.)

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/i18n/languages/russian/countries.ts [hunk 1]
```diff
@@ -156,6 +161,7 @@ export const TUI_RUSSIAN_LANGUAGE_COUNTRIES: Record<TuiCountryIsoCode, string> =
     [TuiCountryIsoCode.PW]: `Палау`,
     [TuiCountryIsoCode.PY]: `Парагвай`,
     [TuiCountryIsoCode.QA]: `Катар`,
+    [TuiCountryIsoCode.RE]: `Встреча`,
```
Comment (splincode) [inline projects/i18n/languages/russian/countries.ts:164]: Реюньон

Comment (VictorienTardif) [inline projects/i18n/languages/russian/countries.ts:164]:
```suggestion
    [TuiCountryIsoCode.RE]: `Реюньон`,
```

File: projects/kit/components/input-phone-international/tokens/countries-masks.ts [hunk 2]
```diff
@@ -70,16 +71,18 @@ export const TUI_COUNTRIES_MASKS: InjectionToken<Record<TuiCountryIsoCode, strin
                 [TuiCountryIsoCode.FJ]: `+679##-#####`,
                 [TuiCountryIsoCode.FK]: `+500#####`,
                 [TuiCountryIsoCode.FM]: `+691###-####`,
-                [TuiCountryIsoCode.FR]: `+33 ## ## ## ## ##`,
+                [TuiCountryIsoCode.FR]: `+33(0) # ## ## ## ##`,
```
Comment (splincode) [inline projects/kit/components/input-phone-international/tokens/countries-masks.ts:74]: why `(0)`

**Comment by splincode** [comment]
@VictorienTardif 
image

File: projects/kit/components/input-phone-international/tokens/countries-masks.ts (hunk 2 above)
Comment (VictorienTardif) [reply projects/kit/components/input-phone-international/tokens/countries-masks.ts:74]: As I said in the PR description ⬆️ : 

French numbers always start with a 0 and although it is valid to omit it when giving the telephone area code, some services do not work without it. This is the case with Firebase, for example, which does not send SMS if the 0 is missing.

`+33612345678` and `+330612345678` are both valid but `+33612345678` does not always work and people tend to omit the 0 when `+33` is displayed so the input displays `+33 61 23 45 67 8`. 
Starting with the 0 and asking for a single character after (which can be either 6 or 7) seems clearer to me.

(+33 0# ## ## ## ## would have been better but this mask does not work while +33(0) # ## ## ## ## does.)

It's an opinionated change that you can dismiss.

Comment (splincode) [reply projects/kit/components/input-phone-international/tokens/countries-masks.ts:74]: thank you

**Review by splincode — APPROVED** [review]

**Review by vladimirpotekhin — APPROVED** [review]

**Comment by splincode** [comment]
@waterplea do you mean?

```ts
getFlagPath(code: TuiCountryIsoCode): string {
        return this.options.getFlagPath?.(code) ?? `${this.staticPath}${code}.png`;
    }
```

**Comment by VictorienTardif** [comment]
Would it be ok for you guys to merge this and handle that on your side in another PR?

**Comment by waterplea** [comment]
@splincode let's not overthink it for the time being.
@VictorienTardif can you just change `getFlagPath` in here:
https://github.com/Tinkoff/taiga-ui/blob/main/projects/kit/components/input-phone-international/input-phone-international.component.ts#L173
to:
```ts
getFlagPath(code: TuiCountryIsoCode): string {
  switch(code) {
    case TuiCountryIsoCode.BL:
    case TuiCountryIsoCode.GF:
    // and all the others
      return `${this.staticPath}FR.png`;
    default: 
      return `${this.staticPath}${code}.png`;
  }
}
```
and remove extra flag icons and I think it's good to merge.

**Review by waterplea — APPROVED** [review]

**Pull request merged**